### PR TITLE
feat: centralize react-query queries 

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

- Keeps things DRY 🩸🛁 and encapsulated in a single-source-of-truth for tidiness and safety 
- Ensures we get the query keys right and don't have to remember them - noticed a few instances of infinitely-cached queries that were wrong and fixed in this diff e.g this guy in develop which should be `pools` (plural) and could cause bugs if we were to do another `midgardPoolData` call somewhere else when expecting a single pool as a response: 
https://github.com/shapeshift/web/blob/deffcaea9271e61a001c141ece3461c903b4025f/src/pages/ThorChainLP/hooks/usePools.ts#L76-L83
- Removes util pure function wrappers over `queryClient.fetchQuery`, since `queryClient.fetchQuery()` calls can now be done in one line 🎉 
- Gives sane `staleTime` defaults to queries - `Infinity` for most, but sometimes a different one when applicable (keeping the same ones as currently), so we don't end up in cases where we void the `staleTime` guarantees by having different consumers use different `staleTime`s

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

Theoretically low to none, effectively medium given the domain diffed. Common THORChain (THORNode/Midgard) fetching queries and THORChain LP queries have been moved to the new source-of-truth. The same queries configuration has been kept so this is most likely very low risk, however, the default `refetchOnMount` setting of `react-query` has been switched to `false` for performance reason, which may produce regressions in lending.

## Testing

- Complete a lending (either deposit or withdraw) flow as a paranoia check and ensure status detection is still happy

### Engineering

- ☝🏽 
- THORChain LP is still happy
- Queries aren't refetched on mount anymore. That's a *global* setting so will apply to any react-query query, not only LP. One easy way to test this against develop or the PR under this one is by going from "Pools" to "Your Positions" and back, which previously triggered a full refetch, whereas this PR now keeps the subscribers alive, which means no refetch 🎉 

### Operations

- ☝🏽 

### Lending paranoia test of status detection

https://github.com/shapeshift/web/assets/17035424/f0932fcb-a0f7-4ade-b935-7604348b806a

### Re-mounted subscribers don't refetch anymore

(this diff DevTools on top, develop on the bottom)
 

https://github.com/shapeshift/web/assets/17035424/24c484b5-614b-4378-a820-50e89ffc4bcf

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6049?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #6028** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6028?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6030** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6030?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #6032** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6032?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #6043** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6043?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #6049** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6049?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6049?utm_source=stack-comment).

File: src/pages/ThorChainLP/queries/hooks/useAllUserLpData.ts [hunk 1]
```diff
@@ -40,34 +34,25 @@ export const useAllUserLpData = ({
   const runeMarketData = useAppSelector(state => selectMarketDataById(state, thorchainAssetId))
 
   const { data: allThornodePools, isSuccess: isThornodePoolsDataLoaded } = useQuery({
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/queries/hooks/useAllUserLpData.ts:36]: update variable to `availableThornodePools` for clarity

**Review by kaladinlight — APPROVED** [review]
beautiful! and here I was going to comment on `feat_blazingly_fast_lp` that we should look at drying up all of those `useQuery` functions :joy:

File: src/pages/ThorChainLP/queries/hooks/useAllUserLpData.ts (hunk 1 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/queries/hooks/useAllUserLpData.ts]: 2a7b7acbec73104067d60468da5ea1d381e6c6b4

**Pull request merged**